aten/src/ATen/native/mps/operations/ReduceOps.mm

```diff
@@ -537,14 +538,13 @@ void impl_func_norm_mps(
           MPSGraphTensor *outputTensor;
 
           if (pIsZero) {
-              MPSGraphTensor *absoluteTensor = [mpsGraph absoluteWithTensor:inputTensor
-                                                                       name:nil];
-              MPSGraphTensor *powerValTensor = [mpsGraph constantWithScalar:p
-                                                                   dataType:mps_input_dtype];
-              MPSGraphTensor *powerTensor = [mpsGraph powerWithPrimaryTensor:absoluteTensor
-                                                             secondaryTensor:powerValTensor
+              MPSGraphTensor* zeros = [mpsGraph constantWithScalar:0.0 dataType:mps_input_dtype];
+              MPSGraphTensor* ones = [mpsGraph constantWithScalar:1.0 dataType:mps_input_dtype];
+              MPSGraphTensor* nonZeros = [mpsGraph selectWithPredicateTensor:inputTensor
+                                                         truePredicateTensor:ones
+                                                        falsePredicateTensor:zeros
```

Comment:
I am actually curious, I would think the select vs absolute with power might be different in perf, too.

Reply:
Previous formula was wrong - I have changed it because it was producing incorrect results. For `ord=0` it should be `sum(x!=0)`. https://numpy.org/doc/stable/reference/generated/numpy.linalg.norm.html, https://pytorch.org/docs/stable/generated/torch.linalg.vector_norm.html#torch.linalg.vector_norm